### PR TITLE
misc

### PR DESCRIPTION
**Pull request by ferncabrera**
Includes:

GlobalBanner timing/styling patches:
- Basically I made sure that we are syncing the timing that the loading bar crawls with the timing that the react component actually shows by using the SCSS variable and some regex. Otherwise we should show the Banner component infinitely and hide the loading bar. So now the "infinite" attribute on the globalBannerState is just a boolean, set it to TRUE if you want the banner to show infinitely, otherwise it will default to the SCSS variable duration of 7s. 

thinking ahead here -> should I make the operations controller that redirects people, hide the any error messages that may be currently showing (reset the globalBannerState?)

### CODE REVIEW

File: microservices/client/src/components/banners/GlobalBanner.tsx [hunk 1]
```diff
@@ -19,14 +20,20 @@ export const GlobalBanner : React.FC<IOpenIMSBannerProps>= ({ customStyleObject
...
-
+    
     const [isSuccess, setIsSuccess] = useState(false);
-
+    
     const memoizedBannerStateData = useMemo(() => bannerStateData, [bannerStateData]);
-
-    const tooltip = (additional) => (
+    //? We want the loading bar to sync with the alert show length, that is set in the varaiables.scss file.
+    let secondsToDisplayAlert = styles.alertDisplaySeconds.match(regex);
```
Comment (Polsander) [inline microservices/client/src/components/banners/GlobalBanner.tsx:29]: Yeah I definitely noticed timing errors before. Good catch and implementation here!

**Review by Polsander — CHANGES_REQUESTED** [review]
Amazing!
Good use of the variables.scss file! Did not think of using it at the time for our first implementation of the global banner.

To address your comment. On "heavy" redirects we do want to reset the banner state, since we do not want to show an error from one page to stay on a different screen... to some extent.

We may want to keep error screens where possible for asynchronous changes. For example, future uploads or notification handling should show across multiple screens until the user dismisses the banner, especially if there is an error. See if you'd like to implement this, if not for now then I am fine with any changes you've made so far merging to main.

**Review by Polsander — APPROVED** [review]
Approved and ready for merge if my comments were addressed :)

**Comment by ferncabrera** [comment]
This was merged into invoices crud - I watched it happen.

**Pull request closed without merging**